test/testdata/desugar/range.rb

```diff
@@ -14,9 +14,39 @@ def foo
     T.reveal_type(e) # error: Revealed type: `Integer`
     T.reveal_type(f) # error: Revealed type: `String`
 
-    # testing for endless ranges
-    g = (1..)
-    h = Range.new(1, nil)
+    # testing for beginless ranges
+    g = (..1)
+    h = Range.new(nil, 1)
     T.reveal_type(g) # error: Revealed type: `T::Range[Integer(1)]`
     T.reveal_type(h) # error: Revealed type: `T::Range[Integer(1)]`
+
+    # testing for endless ranges
+    i = (1..)
+    j = Range.new(1, nil)
+    T.reveal_type(i) # error: Revealed type: `T::Range[Integer(1)]`
+    T.reveal_type(j) # error: Revealed type: `T::Range[Integer(1)]`
+
+    # testing nil..nil edge case
+    k = (nil..)
+    l = Range.new(nil, nil)
+    T.reveal_type(k) # error: Revealed type: `T::Range[NilClass]`
+    T.reveal_type(l) # error: Revealed type: `T::Range[NilClass]`
```

Comment:
I think both of these should be `T::Range[T.untyped]` since we can't say anything about the entries in the range. The fact that the boundaries are `nil` does not mean that all the entries are `nil`, it just means there are no boundaries.

Reply:
Can it?

Do you refer to things like `T::Range[Integer].new(nil, nil)`? On both 2.6 and 2.7 I was unable to get any other value out of this kind of range but `nil`.

Reply:
Here is why I think this might lead to type problems. If I have a database query interface that works with ranges, I can define it like this:
```ruby
class Model
  sig { params(id_range: T::Range[Integer]).returns(T::Array[Model]) }
  def self.find(id_range)
    # do some querying
  end
end
```

with the intention of it being used in the following ways:
```ruby
# Query a range of ids
Model.find(1..3) #=> [Model(1), Model(2), Model(3)]

# Query ids after 5
Model.find(5..) #=> [Model(5), Model(6)]

# Query ids up to 2
Model.find(..2) #=> [Model(1), Model(2)]

# Query all records
Model.find(Range.new(nil, nil)) #=> [Model(1), Model(2), Model(3), Model(4), Model(5), Model(6)]
```

However, if Sorbet infers the type of `Range.new(nil, nil)` as `T::Range[NilClass]` the last call above will not type-check since `T::Range[NilClass]` will not be compatible with `T::Range[Integer]`. But that does not sound right!

That is why I am claiming that (regardless of what items we can get out of it), we should treat unbounded ranges as items that we have no idea what is inside, so that it is compatible with all interfaces. 

Thus, I propose this should be `T::Range[T.untyped]`

Reply:
I agree, and I think that with the changes that @Morriar and I discussed last week, it will be.